src/sortStudents.ts

```diff
@@ -1,16 +1,80 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  function averGrade(arr: number[]): number {
+    const grade = arr.reduce((sum: number, a: number):
+    number => sum + a, 0) / arr.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    return grade;
+  }
+
+  const newStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return order === 'asc'
+        ? newStudents.sort((
+          a: Student,
+          b: Student,
+        ):
+        number => a[sortBy].localeCompare(b[sortBy]))
+        : newStudents.sort((a: Student, b: Student):
+        number => b[sortBy].localeCompare(a[sortBy]));
+
+    case SortType.Age:
+      return order === 'asc'
+        ? newStudents.sort((a: Student, b: Student):
+        number => a.age - b.age)
+        : newStudents.sort((a: Student, b: Student):
+        number => b.age - a.age);
+
+    case SortType.Married:
+      newStudents.sort((a: Student, b: Student): number => {
+        if (a.married && !b.married) {
```

Comment:
Also, you can combine it with age )

```suggestion
        return order === 'asc'
          ? +(a[sortBy]) - +(b[sortBy])
          : +(b[sortBy]) - +(a[sortBy]);
```